### PR TITLE
[Enhancement] protect from dict code out of range

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/formats/parquet/encoding_dict.h [hunk 1]
```diff
@@ -221,6 +221,13 @@ class DictDecoder<Slice> final : public Decoder {
...
         case VALUE: {
             raw::stl_vector_resize_uninitialized(&_slices, count);
+            size_t size = _dict.size();
+            for (int i = 0; i < count; ++i) {
+                if (_indexes[i] >= size) {
+                    return Status::InternalError(
+                            fmt::format("dict code is out of range. code = {}, size = {}", _indexes[i], size));
+                }
+            }
             for (int i = 0; i < count; ++i) {
```
Comment (GavinMar) [inline be/src/formats/parquet/encoding_dict.h:231]: Why not check `_indexes[i]` before this line and avoid introducing an extra for-loop?

Comment (dirtysalt) [reply be/src/formats/parquet/encoding_dict.h:231]: to avoid branch on decode stage.

**Review by Smith-Cruise — DISMISSED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by zombee0 — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**